policies/recipes/steps.go

```diff
@@ -127,106 +126,79 @@ func StepFileExec(step *osconfigpb.SoftwareRecipe_Step_FileExec, artifacts map[s
 		return fmt.Errorf("can't determine location type")
 	}
 
-	return executeCommand(path, stepDir, runEnvs, step.FileExec.Args...)
+	return executeCommand(path, step.FileExec.Args, stepDir, runEnvs, []int32{0})
 }
 
-// StepScriptRun builds the command for a ScriptRun step
+// StepScriptRun runs a ScriptRun step.
 func StepScriptRun(step *osconfigpb.SoftwareRecipe_Step_ScriptRun, artifacts map[string]string, runEnvs []string, stepDir string) error {
+	cmd := filepath.Join(stepDir, "recipe_script_source")
+	if err := writeScript(cmd, step.ScriptRun.Script); err != nil {
+		return err
+	}
+	args := step.ScriptRun.Args
 	switch step.ScriptRun.Interpreter {
 	case osconfigpb.SoftwareRecipe_Step_RunScript_INTERPRETER_UNSPECIFIED:
-		return scriptRunDirect(step, runEnvs, stepDir)
+		if runtime.GOOS == "windows" {
+			args = append([]string{"/c", cmd}, args...)
+			cmd = "C:\\Windows\\System32\\cmd.exe"
+		}
 	case osconfigpb.SoftwareRecipe_Step_RunScript_SHELL:
 		if runtime.GOOS == "windows" {
-			return scriptRunCmd(step, runEnvs, stepDir)
+			args = append([]string{"/c", cmd}, args...)
+			cmd = "C:\\Windows\\System32\\cmd.exe"
+		}
+		qargs := cmd
+		for _, arg := range args {
+			qargs = fmt.Sprintf("%s %q", qargs, arg)
```

Comment:
the end result here has to be ["/bin/sh", "-c", "/path/to/recipe_script_source \"quoted arg\" \"quoted arg\""]; args have to end up with script name in one string at the end. unsure if cmd needs something similar, will investigate